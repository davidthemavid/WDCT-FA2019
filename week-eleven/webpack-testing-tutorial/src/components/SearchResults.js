import React from "react";

export default function SearchResults({ articles = [] }) {
  return (
    <ul>
      {articles.map(({ webUrl, webTitle }) => (
        <li key={webUrl}>
          <a href={webUrl} target="_blank">
            {webTitle}
          </a>
        </li>
      ))}
    </ul>
  );
}
