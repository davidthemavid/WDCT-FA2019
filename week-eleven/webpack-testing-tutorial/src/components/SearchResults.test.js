import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SearchResults from "../components/SearchResults";

const mockArticles = [
  {
    webUrl: "http://www.google.ca",
    webTitle: "This is Google's HomePage"
  }
];
// configure enzyme for react components
Enzyme.configure({ adapter: new Adapter() });

// suite or collection of tests
describe("Testing SearchResults Component", () => {
  test("returns empty array when no props are passed", () => {
    const wrapper = shallow(<SearchResults />);
    // assertions
    expect(wrapper).toMatchSnapshot();
  });
  test("no <li> elements when no props are passed", () => {
    const wrapper = shallow(<SearchResults />);
    expect(wrapper.find("li")).toHaveLength(0);
  });
  test("renders one article", () => {
    const wrapper = mount(<SearchResults articles={mockArticles} />);
    expect(wrapper.find("li")).toHaveLength(1);
  });
  test("renders one article", () => {
    const wrapper = mount(<SearchResults articles={[]} />);
    expect(wrapper.find("li")).toHaveLength(0);
  });
});
