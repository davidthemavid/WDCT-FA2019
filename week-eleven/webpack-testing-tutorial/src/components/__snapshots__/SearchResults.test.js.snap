// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchResults Component returns empty array when no props are passed 1`] = `ShallowWrapper {}`;
